data/World/Ganons Castle.json

```diff
@@ -78,7 +87,27 @@
                 Hover_Boots or can_play(Song_of_Time)",
             "Ganons Castle Shadow Trial Golden Gauntlets Chest": "
                 can_use(Fire_Arrows) or 
-                (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire)))"
+                (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire)))",
+            "Ganons Castle Shadow Trial Recovery Heart 1": 
+                "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
+            "Ganons Castle Shadow Trial Recovery Heart 2": 
+                    "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
+            "Ganons Castle Shadow Trial Recovery Heart 3": 
+                    "(logic_lens_castle or can_use(Lens_of_Truth)) and (
+                    can_use(Fire_Arrows) or 
+                    (can_use(Longshot) and (Hover_Boots or can_use(Dins_Fire))))",
```

Comment:
You can reach each of these hearts with the Hovers Boots, without any knowledge of where the invisible platforms are. Therefore, when using the Hover Boots only, these need not check for Lens/Trick. (Notice how clearing the trial also does not require lens when using the Longshot/Hovers combo. Unlike clearing the trial though, Fire Arrows -can- combo with Hovers to skip the Lens, so you could just offer Hovers as an alternative to Lens then I think)